src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -619,4 +620,31 @@ public Charset detectCharset(byte[] contentArray) {
 
         return Charset.forName(charsetName);
     }
+
+    private Map<Path, ScheduledFuture> futures = new HashMap<>();
+
+    private ScheduledExecutorService executor = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
+
+    private static final TimeUnit MINUTES = TimeUnit.MINUTES; // your time unit
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)